tests/Microsoft.Identity.Test.Integration.netcore/HeadlessTests/FmiIntegrationTests.cs

```diff
@@ -0,0 +1,441 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Test.LabInfrastructure;
+using Microsoft.Identity.Test.Unit;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.Identity.Test.Common.Core.Helpers;
+using System;
+using System.IdentityModel.Tokens.Jwt;
+
+namespace Microsoft.Identity.Test.Integration.NetCore.HeadlessTests
+{
+    /// <summary>
+    /// The tests in this file are demonstrations of the various authentication flows outlined in the "FMI protocol spec v1.0" Section 3.2
+    /// </summary>
+    [TestClass]
+    public class FmiIntegrationTests
+    {
+        private byte[] _serializedCache;
+
+        [TestMethod]
+        //RMA getting FMI cred for a leaf entity or sub-RMA
+        public async Task Flow1_RmaCredential_From_CertTestAsync()
+        {
+            //Arrange
+            X509Certificate2 cert = CertificateHelper.FindCertificateByName(TestConstants.AutomationTestCertName);
+            string expectedExternalCacheKey = null;
+
+            Action<TokenCacheNotificationArgs> extCacheKeyEvaluator = (args) =>
+            {
+                if (expectedExternalCacheKey != null)
+                {
+                    Assert.IsTrue(args.SuggestedCacheKey.Contains(expectedExternalCacheKey));
+                }
+            };
+
+            //Fmi app/scenario parameters
+            var clientId = "4df2cbbb-8612-49c1-87c8-f334d6d065ad";
+            var scope = "api://AzureFMITokenExchange/.default";
```

Comment:
nit: no reason for this variable.

Reply:
I kinda like them for the things that change the most in the tests. Will help others clearly see what is different